Sources/ShellOut.swift

```diff
@@ -416,6 +444,91 @@ private extension Process {
             return outputData.shellOutput()
         }
     }
+    
+    func launchBash(with outputHandle: FileHandle? = nil, errorHandle: FileHandle? = nil, withCompletion completion: @escaping Completion) {
+        
+        var outputData = Data()
+        var errorData = Data()
+        
+        let outputPipe = Pipe()
+        standardOutput = outputPipe
+        
+        let errorPipe = Pipe()
+        standardError = errorPipe
+        
+        // Because FileHandle's readabilityHandler might be called from a
+        // different queue from the calling queue, avoid a data race by
+        // protecting reads and writes to outputData and errorData on
+        // a single dispatch queue.
+        let outputQueue = DispatchQueue(label: "bash-output-queue")
+        
+        #if !os(Linux)
+        outputPipe.fileHandleForReading.readabilityHandler = { handler in
+            outputQueue.async {
+                let data = handler.availableData
+                outputData.append(data)
+                DispatchQueue.main.async {
+                    outputHandle?.write(data)
+                }
+            }
+        }
+        
+        errorPipe.fileHandleForReading.readabilityHandler = { handler in
+            outputQueue.async {
+                let data = handler.availableData
+                errorData.append(data)
+                DispatchQueue.main.async {
+                    errorHandle?.write(data)
```

Comment:
The `errorHandle` reference falls into the same concerns we have with `outputHandle`. 

See above comment.